### PR TITLE
feat(ci): use sccache with DO space

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

-

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: .github/workflows/forest.yml [hunk 1]
```diff
@@ -44,6 +48,9 @@ jobs:
     name: Build Ubuntu (Forest daemon)
     runs-on: ubuntu-latest
     steps:
+      - name: Show IP
```
Comment (lemmih) [inline .github/workflows/forest.yml:51]: Consider adding a comment here saying this aims to find the nearest S3 space for storing cache files.

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -24,7 +24,11 @@ env:
   CI: 1
   CARGO_INCREMENTAL: 0
   CACHE_TIMEOUT_MINUTES: 5
-  SCCACHE_GHA_ENABLED: "true"
+  SCCACHE_ENDPOINT: "https://sfo3.digitaloceanspaces.com"
+  SCCACHE_BUCKET: "forest-sccache-us-west"
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply .github/workflows/unit-tests.yml:28]: @LesnyRumcajs  It's possible but requires admin permission. @lemmih what do u think?

https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

File: .github/workflows/forest.yml (hunk 1 above)
Comment (hanabi1224) [reply .github/workflows/forest.yml:52]: fixed

File: .github/workflows/unit-tests.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lemmih) [reply .github/workflows/unit-tests.yml:28]: I added `SCCACHE_ENDPOINT`, `SCCACHE_BUCKET` and `SCCACHE_REGION` as repository wide environment variables. You should be able to remove those from the workflows now. You still need to set the access key and secret key.

[the remark this answers is not in the document]
Comment (hanabi1224) [reply .github/workflows/unit-tests.yml]: updated

**Pull request merged**